File: src/main/java/com/autosale/shop/service/ProductService.java
package com.autosale.shop.service;

import com.autosale.shop.model.PaginationRequest;
import com.autosale.shop.model.PaginationResponse;
import com.autosale.shop.model.Product;
import com.autosale.shop.model.ProductStatus;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface ProductService {

    PaginationResponse<Product> findByStatus(PaginationRequest paginationRequest, ProductStatus status);

    Product findById(int id);

    List<Product> findAllFromCurrentUser();

    List<Product> findAllByUserId(int id);

    int create(Product product);

    void edit(Product product);

    int deleteById(int id);

    void buy(int productId);
}

File: src/main/java/com/autosale/shop/service/ProductService.kt
package com.autosale.shop.service

import com.autosale.shop.model.PaginationRequest
import com.autosale.shop.model.PaginationResponse
import com.autosale.shop.model.Product
import com.autosale.shop.model.ProductStatus
import org.springframework.stereotype.Service

@Service
interface ProductService {
    fun findByStatus(paginationRequest: PaginationRequest, status: ProductStatus?): PaginationResponse<Product>
    fun findById(id: Int): Product
    fun findAllFromCurrentUser(): List<Product>
    fun findAllByUserId(id: Int): List<Product>
    fun create(product: Product): Int
    fun edit(product: Product)
    fun deleteById(id: Int): Int
    fun buy(productId: Int)
    fun countAllActive(): Int
}
